pages/book/contracts.mdx

```diff
@@ -0,0 +1,203 @@
+
+</Callout>
+
+### Contract constants [#constants]
+
+Unlike [variables](#variables), constants cannot change. Their values are calculated in _compile-time_ and cannot change during execution.
+
+There isn't much difference between constants defined outside of a contract (global constants) and inside the contract (contract constants). Those defined outside can be used by other contracts in your project.
+
+Constant initializations must be relatively simple and only rely on values known during compilation. If you add two numbers for example, the compiler will calculate the result during build and put the result in your compiled code.
+
+You can read constants both in [receivers](#receiver-functions) and in [getters](#getter-functions).
+
+Unlike [contract variables](#variables), **contract constants don't consume space in persistent state**. Their values are stored directly in the code [`Cell{:tact}`](/book/types#primitive-types) of the contract.
+
+```tact
+// global constants are calculared in compile-time and cannot change
+const GlobalConst1: Int = 1000 + ton("42") + pow(10, 9);
+
+contract Example {
+    // contract constants are also calculated in compile-time and cannot change
+    const ContractConst1: Int = 2000 + ton("43") pow(10, 9);
+
+    // contract constants can be an easy alternative to enums
+    const StateUnpaid: Int = 0;
+    const StatePaid: Int = 1;
+    const StateDelivered: Int = 2;
+    const StateDisputed: Int = 3;
+
+    // no need to init constants
+    init() {}
+
+    get fun sum(): Int {
+        // access constants from anywhere
+        return GlobalConst1 + self.ContractConst1 + self.StatePaid;
+    }
+}
+```
+
+Read more about constants on their dedicated page: [Constants](/book/constants).
+
+### Constructor function `init()` [#init-function]
+
+On deployment of the contract, the constructor function `init(){:tact}` is run.
+
+If a contract has any [persistent state variables](#variables) without default values specified, it must initialize them in this function.
+
+As stated previously, [contract constants](#constants) don't need to be initialized in the `init{:tact}` function, as they get their values in _compile-time_ and don't consume space in persistent state.
+
+```tact
+contract Example {
+    // persistent state variables
+    var1: Int = 0; // initialized with default value 0
+    var2: Int;     // must be initialized in the init() function
+
+    // constructor function
+    init() {
+        self.var2 = 42;
+    }
+}
+```
+
+<Callout>
+
+  To obtain initial state of the target contract in [internal functions](#internal-functions), [receivers](#receiver-functions) or [getters](#getter-functions) use [`initOf{:tact}`](/book/expressions#initof) expression.
+
+</Callout>
+
+### Getter functions
+
+Getter functions are **not accessible from other contracts and exported only to off-chain world**:
```

Comment:
Neither this section, not the linked section explain that getters cannot modify the contract's state variables